app/scripts/app.js

```diff
@@ -6,12 +6,22 @@ var app = angular.module('BlocChat', [
 
   app.controller('HomeController', ['$scope', 'Room', function($scope, Room) {
 
+    var currentRoom = null;
     // get list of rooms from factory
     $scope.rooms = Room.all;
+    debugger;
 
+    $scope.selectRoom = function(roomId) {
+      currentRoom = Room.getRoomById(roomId);
+      parseMessages(currentRoom);
+    }
+
+    parseMessages = function(currentRoom) {
+      return currentRoom.val();
```

Comment:
This is because you're calling `.val()` on the `on` return value.  You should instead change the `getRoomById` method to actually return the room and create a function to get the messages from it using a callback:

Something like:
```js
// inside controller
/* snip */
var currentRoom = null;

$scope.selectRoom = function (roomId) {
  currentRoom = Room.getRoomById(roomId);
  Room.parseMessages(currentRoom, function (messages) {
    // do something with the messages
  });
};
/* snip */

// inside Room
/* snip */
getRoomById: function (id) {
  // replace "url" with whatever the URL is
  return new Firebase(url);
},
parseMessages: function (room, callback) {
  room.child("messages").on("value", callback);
}
/* snip */
```